bittide-instances/tests/Wishbone/ScatterGather.hs

```diff
@@ -0,0 +1,111 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE NumericUnderscores #-}
+{-# OPTIONS_GHC -fplugin=Protocols.Plugin #-}
+
+module Wishbone.ScatterGather where
+
+import Clash.Explicit.Prelude
+import Clash.Prelude (HiddenClockResetEnable, withClockResetEnable)
+
+import Data.Char (chr)
+import Data.Maybe (mapMaybe)
+import Language.Haskell.TH
+import Project.FilePath
+import Protocols
+import Protocols.Idle
+import System.FilePath
+import Test.Tasty
+import Test.Tasty.HUnit
+import Test.Tasty.TH
+import VexRiscv (DumpVcd (NoDumpVcd))
+
+import Bittide.Calendar
+import Bittide.DoubleBufferedRam
+import Bittide.ProcessingElement
+import Bittide.ProcessingElement.Util
+import Bittide.ScatterGather
+import Bittide.SharedTypes
+import Bittide.Wishbone
+
+import qualified Protocols.Df as Df
+import qualified Prelude as P
+
+case_scatter_gather_echo_test :: Assertion
+case_scatter_gather_echo_test = do
+  assertBool
+    msg
+    (P.head (lines simResult) == "Written data was read back correctly")
+ where
+  msg =
+    "Received the following from the CPU over UART:\n"
+      <> unlines (P.take 5 (lines simResult))
```

Comment:
Why take 5? If you want to limit the maximum execution time of the test or prevent timeout, you should probably do so by manipulating the configuration of `sampleC` instead.